gremlin-go/driver/driverRemoteConnection.go

```diff
@@ -19,17 +19,51 @@ under the License.
 
 package gremlingo
 
-import "golang.org/x/text/language"
+import (
+	"crypto/tls"
+	"golang.org/x/text/language"
+	"net/http"
+)
+
+// AuthInfo is an option struct that allows authentication information to be specified.
+// Authentication can be provided via http.Header Header directly.
+// Basic authentication can also be used via the BasicAuthInfo function.
+type AuthInfo struct {
+	Header   http.Header
+	Username string
+	Password string
+}
+
+// getHeader provides a safe way to get a header from the AuthInfo even if it is nil.
+// This way we don't need any additional logic in the transport layer.
```

Comment:
Do we still need this function? It seems like a safeguard against bad practices, and given that the constructor is moved to a settings object where the settings has a default non-nil value for the AuthInfo I don't believe this is needed.